apps/condo/domains/resident/tasks/notifyResidentsOnPayday.js

```diff
@@ -0,0 +1,144 @@
+const { sendMessage, Message } = require('@condo/domains/notification/utils/serverSchema')
+const { ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+
+
+
+const logger = getLogger('notifyResidentsOnPayday')
+
+const DV_SENDER = { dv: 1, sender: { dv: 1, fingerprint: 'notifyResidentsOnPayday' } }
+const conflictingMessageTypes = [BILLING_RECEIPT_ADDED_TYPE, BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE, BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE, BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE, BILLING_RECEIPT_AVAILABLE_TYPE, BILLING_RECEIPT_CATEGORY_AVAILABLE_TYPE, SEND_BILLING_RECEIPTS_ON_PAYDAY_REMINDER_MESSAGE_TYPE]
+
+async function hasConflictingPushes (context, userId) {
+    const now = dayjs()
+    const userMessages = await Message.getAll(context, {
+        user: { id: userId },
+        createdAt_gte: now.startOf('date').toISOString(),
+        createdAt_lte: now.toISOString(),
+        deletedAt: null,
+    })
+
+    for (const message of userMessages) {
+        if (conflictingMessageTypes.find(type => type === message.type)) return true
+    }
+    return false
+}
+
+async function sendNotification (context, receipt, consumer) {
+    const resident = consumer.resident
+    const userId = resident.user.id
+    const lang = get(COUNTRIES, [get(receipt, 'organization.country', conf.DEFAULT_LOCALE), 'locale'], conf.DEFAULT_LOCALE)
+    const now = dayjs().subtract(1, 'month')
+
+    if (await hasConflictingPushes(context, userId)) {
+        logger.info({ msg: 'Today the User received push notifications with a conflicting type', userId })
+        return
+    }
+
+    const uniqKey = `${userId}_notification_on_payday_${receipt.period}`
+    await sendMessage(context, {
+        ...DV_SENDER,
+        to: { user: { id: userId } },
+        lang,
+        type: SEND_BILLING_RECEIPTS_ON_PAYDAY_REMINDER_MESSAGE_TYPE,
+        meta: {
+            dv: 1,
+            data: {
+                monthName: lang === RU_LOCALE ? now.locale('ru').format('MMMM') : now.format('MMMM'),
+                serviceConsumerId: consumer.id,
+                residentId: resident.id,
+                userId,
+            },
+        },
+        uniqKey,
+    })
+}
+
+async function notifyResidentsOnPayday () {
+    const { keystone: context } = await getSchemaCtx('User')
+    const state = {
+        startTime: dayjs(),
+        hasMoreConsumers: true,
+        consumersOffset: 0,
+        consumersChunkSize: 50,
+        processedReceipts: 0,
+        recipientsMap: new Map(),
+        failedConsumers: 0,
+        recipients: [],
+    }
+    logger.info({ msg: 'Start processing', startAt: state.startTime.format() })
```

Comment:
Use task name or something your task specific. It’s hard to connect this message with the function.

Reply:
It looks like useless message …